### PR TITLE
refactor UserSearchResultsViewControllerTests

### PR DESCRIPTION
**Pull request by billypchan**
## What's new in this PR?

Refactor `UserSearchResultsViewControllerTests` with Swift snapshot testing and prevent creating `ZMUser`s

### CODE REVIEW

File: Wire-iOS Tests/UserSearchResultsViewControllerTests.swift [hunk 1]
```diff
@@ -19,77 +19,69 @@
...
-
-        for name in usernames {
-            let user = ZMUser.insertNewObject(in: uiMOC)
-            user.remoteIdentifier = UUID()
-            user.teamIdentifier = nil
-            user.name = name
-            user.setHandle(name.lowercased())
+        var allUsers: [UserType] = []
+
+		for name in XCTestCase.usernames {
```
Comment (johnxnguyen) [inline Wire-iOS Tests/UserSearchResultsViewControllerTests.swift:81]: Where is `XCTestCase.usernames` defined? Why is it an extension on `XCTestCase`?

**Review by johnxnguyen — CHANGES_REQUESTED** [review]
There are quite a few issues with indentation, looks like your editor is using 2 spaces.

File: Wire-iOS Tests/UserSearchResultsViewControllerTests.swift (hunk 1 above)
Comment (billypchan) [reply Wire-iOS Tests/UserSearchResultsViewControllerTests.swift:81]: `usernames` was a property `CoreDataSnapshotTestCase`. I put it on its super class `XCTestCase` after refactoring. We can also put it in another class like `MockUserType`

Comment (johnxnguyen) [reply Wire-iOS Tests/UserSearchResultsViewControllerTests.swift]: I think `MockUserType` is a more suitable place to put it.

**Review by johnxnguyen — APPROVED** [review]

**Review by marcomaddy1982 — APPROVED** [review]

**Pull request merged**